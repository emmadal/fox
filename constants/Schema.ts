import { email, username } from "./Regex";
import i18n from "@/i18n";
import { z } from "zod";

export const loginSchema = z
  .object({
    email: z
      .string({ required_error: `${i18n.t("fieldrequired")}` })
      .regex(email, { message: `${i18n.t("invalidemail")}` })
      .trim(),
    password: z
      .string({ required_error: `${i18n.t("fieldrequired")}` })
      .min(5, `${i18n.t("passwordmin")}`)
      .trim(),
  })
  .required();

export const registerSchema = z
  .object({
    email: z
      .string({ required_error: `${i18n.t("fieldrequired")}` })
      .email({ message: `${i18n.t("invalidemail")}` })
      .trim(),
    name: z
      .string({ required_error: `${i18n.t("fieldrequired")}` })
      .max(50)
      .trim(),
    username: z
      .string({ required_error: `${i18n.t("fieldrequired")}` })
      .regex(username, { message: `${i18n.t("usernameinvalid")}` })
      .max(20)
      .trim(),
    password: z
      .string({ required_error: `${i18n.t("fieldrequired")}` })
      .min(5, `${i18n.t("passwordmin")}`)
      .trim(),
  })
  .required();

export const resetPasswordSchema = z.object({
  email: z
    .string({ required_error: `${i18n.t("fieldrequired")}` })
    .email()
    .trim(),
});

export const editProfileSchema = z.object({
  name: z
    .string({ required_error: `${i18n.t("fieldrequired")}` })
    .max(50)
    .trim(),
  biography: z.string().max(200).trim().optional(),
  avatar_url: z.string().url().optional(),
  location: z.string().max(50).trim().optional(),
  website: z.string().url().trim().optional(),
});
