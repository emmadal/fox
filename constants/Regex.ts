/* eslint-disable no-useless-escape */
/*
 * username validation
 * accepted: letters & numbers, without space
 */
export const username = /^([a-zA-Z0-9\-_])*[^\s]\1*$/;
/*
 * email validation
 */
export const email = /^[^\s@]+@[^\s@]+\.([^\s@]{2,})+$/;

export const number = /[0-9]|\./;

export const phone =
  /^(?:\+\d{1,2}\s*)?(?:\(?\d{3}\)?[\s.-]?)?\d{3}[\s.-]?\d{4}$/;

/*
 * password validation, should contain:
 * (?=.*\d): at least one digit
 * (?=.*[a-z]): at least one lower case
 * (?=.*[A-Z]): at least one uppercase case
 * [0-9a-zA-Z]{8,}: at least 8 from the mentioned characters
 * (?=.*[!@#$%^&*]): at least one special characters
 */
export const password = /^(?=.*\d)(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}$/;
