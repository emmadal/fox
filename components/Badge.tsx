import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { useColorScheme } from "@/hooks/useColorScheme";
import { Colors } from "@/constants/Colors";

interface BadgeProps {
  text: string;
  color?: "primary" | "secondary" | "success" | "danger";
}

export const Badge: React.FC<BadgeProps> = ({ text, color = "primary" }) => {
  const colorScheme = useColorScheme();
  const isDark = colorScheme === "dark";

  const getBadgeColor = () => {
    const baseColor = isDark ? Colors.dark : Colors.light;
    switch (color) {
      case "primary":
        return baseColor.tint;
      case "secondary":
        return baseColor.tabIconDefault;
      case "success":
        return "#4caf50";
      case "danger":
        return "#f44336";
      default:
        return baseColor.tint;
    }
  };

  return (
    <View
      style={[styles.badge, { backgroundColor: getBadgeColor() }]}
      testID="badge"
    >
      <Text className="text-center text-xs text-white">{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  badge: {
    borderRadius: 50,
    height: 22,
    width: 22,
    position: "absolute",
    top: -10,
    right: -7,
    zIndex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
});
