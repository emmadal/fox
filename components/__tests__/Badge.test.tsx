import React from "react";
import { render } from "@testing-library/react-native";
import { Badge } from "../Badge"; // Adjust the path to your Badge component
import { useColorScheme } from "@/hooks/useColorScheme";
import { Colors } from "@/constants/Colors";

// Mocking `useColorScheme`
jest.mock("@/hooks/useColorScheme", () => ({
  useColorScheme: jest.fn(),
}));

describe("Badge Component", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<Badge text="1" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("displays the text correctly", () => {
    const { getByText } = render(<Badge text="1" />);
    expect(getByText("1")).toBeTruthy();
  });

  it("applies the primary color in light mode by default", () => {
    (useColorScheme as jest.Mock).mockReturnValue("light");

    const { getByTestId } = render(<Badge text="1" />);
    const badgeStyle = getByTestId("badge").props.style;

    // Replace 'Colors.light.tint' with the expected color
    expect(badgeStyle).toContainEqual({ backgroundColor: Colors.light.tint });
  });

  it("applies the secondary color in dark mode", () => {
    (useColorScheme as jest.Mock).mockReturnValue("dark");

    const { getByTestId } = render(<Badge text="2" color="secondary" />);
    const badgeStyle = getByTestId("badge").props.style;

    // Replace 'Colors.dark.tabIconDefault' with the expected color
    expect(badgeStyle).toContainEqual({
      backgroundColor: Colors.dark.tabIconDefault,
    });
  });

  it('applies the correct color for "success"', () => {
    const { getByTestId } = render(<Badge text="3" color="success" />);
    const badgeStyle = getByTestId("badge").props.style;

    expect(badgeStyle).toContainEqual({ backgroundColor: "#4caf50" });
  });

  it('applies the correct color for "danger"', () => {
    const { getByTestId } = render(<Badge text="4" color="danger" />);
    const badgeStyle = getByTestId("badge").props.style;

    expect(badgeStyle).toContainEqual({ backgroundColor: "#f44336" });
  });
});
