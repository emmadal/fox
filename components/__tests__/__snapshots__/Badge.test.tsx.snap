// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 50,
        "height": 22,
        "justifyContent": "center",
        "position": "absolute",
        "right": -7,
        "top": -10,
        "width": 22,
        "zIndex": 1,
      },
      {
        "backgroundColor": "#0a7ea4",
      },
    ]
  }
  testID="badge"
>
  <Text
    className="text-center text-xs text-white"
  >
    1
  </Text>
</View>
`;
